src/EnergyPlus/WaterToWaterHeatPumpEIR.cc

```diff
@@ -0,0 +1,793 @@
+
+            this->setRunStateAndFlowRates(RunFlag);
+            this->loadSideInletTemp = DataLoopNode::Node(this->loadSideNodes.inlet).Temp;
+            this->sourceSideInletTemp = DataLoopNode::Node(this->sourceSideNodes.inlet).Temp;
+
+            if (calledFromLocation.loopNum == this->sourceSideLocation.loopNum) { // condenser side
+                PlantUtilities::UpdateChillerComponentCondenserSide(this->sourceSideLocation.loopNum,
+                                                                    this->sourceSideLocation.loopSideNum,
+                                                                    this->plantTypeOfNum,
+                                                                    this->sourceSideNodes.inlet,
+                                                                    this->sourceSideNodes.outlet,
+                                                                    this->sourceSideHeatTransfer,
+                                                                    this->sourceSideInletTemp,
+                                                                    this->sourceSideOutletTemp,
+                                                                    this->sourceSideMassFlowRate,
+                                                                    FirstHVACIteration);
+                return;
+            }
+
+            if (this->running) {
+
+                Real64 loadSideOutletSetpointTemp = this->getLoadSideOutletSetpointTemp();
+                Real64 const sourceInletNodeTemp = DataLoopNode::Node(this->sourceSideNodes.inlet).Temp;
+                Real64 capacityModifierFuncTemp = CurveManager::CurveValue(
+                        this->capFuncTempCurveIndex, loadSideOutletSetpointTemp, sourceInletNodeTemp
+                );
+                Real64 availableCapacity = this->referenceCapacity * capacityModifierFuncTemp;
+                Real64 partLoadRatio = 0.0;
+                if (availableCapacity > 0) {
+                    partLoadRatio = max(0.0, min(std::abs(CurLoad) / availableCapacity, 1.0));
+                }
+
+                auto &thisLoadPlantLoop = DataPlant::PlantLoop(this->loadSideLocation.loopNum);
+                Real64 Cp = FluidProperties::GetSpecificHeatGlycol(
+                        thisLoadPlantLoop.FluidName,
+                        DataLoopNode::Node(this->loadSideNodes.inlet).Temp,
+                        thisLoadPlantLoop.FluidIndex,
+                        "WWHPEIR::simulate()"
+                );
+
+                this->loadSideHeatTransfer = availableCapacity * partLoadRatio;
+
+                if (this->plantTypeOfNum == DataPlant::TypeOf_HeatPumpEIRHeating) {
+
+                    // calculate load side
+                    Real64 const loadMCp = this->loadSideMassFlowRate * Cp;
+                    this->loadSideOutletTemp = this->loadSideInletTemp + this->loadSideHeatTransfer / loadMCp;
```

Comment:
These (heating and cooling) seem like they need protection from loadSideMassFlowRate = 0. What if the IF and ELSEIF statements were:

    if (this->plantTypeOfNum == DataPlant::TypeOf_HeatPumpEIRHeating && this->loadSideMassFlowRate > 0.0) {

Reply:
Or maybe flow must be > 0 if code makes it to this point.

Reply:
@rraustad I'm sure you are right.  I need to pull a good chunk of protection from other models before this is wrapped up.  Right now I'm trying to get a completely functioning model, then I'll go through and add robustness and other additions.